core/src/main/scala/kafka/network/RequestConvertToJson.scala

```diff
@@ -0,0 +1,343 @@
+    node.set("connection", new TextNode(context.connectionId))
+    node.set("totalTime", new DoubleNode(totalTimeMs))
+    node.set("requestQueueTime", new DoubleNode(requestQueueTimeMs))
+    node.set("localTime", new DoubleNode(apiLocalTimeMs))
+    node.set("remoteTime", new DoubleNode(apiRemoteTimeMs))
+    node.set("throttleTime", new LongNode(apiThrottleTimeMs))
+    node.set("responseQueueTime", new DoubleNode(responseQueueTimeMs))
+    node.set("sendTime", new DoubleNode(responseSendTimeMs))
+    node.set("securityProtocol", new TextNode(context.securityProtocol.toString))
+    node.set("principal", new TextNode(session.principal.toString))
+    node.set("listener", new TextNode(context.listenerName.value))
+    node.set("clientInformation", new TextNode(context.clientInformation.toString))
+    if (temporaryMemoryBytes > 0)
+      node.set("temporaryMemoryBytes", new LongNode(temporaryMemoryBytes))
+    if (messageConversionsTimeMs > 0)
+      node.set("messageConversionsTime", new DoubleNode(messageConversionsTimeMs))
+    node
+  }
+
+  def requestDesc(header: RequestHeader, req: AbstractRequest, verbose: Boolean): JsonNode = {
+    val node = new ObjectNode(JsonNodeFactory.instance)
+    node.set("requestHeader", requestHeaderNode(header))
+    node.set("request", request(req, verbose))
+    node
+  }
+
+  /**
+   * Temporary until switch to use the generated schemas.
+   */
+  def offsetsForLeaderEpochRequestNode(request: OffsetsForLeaderEpochRequest, version: Short): JsonNode = {
+    val node = new ObjectNode(JsonNodeFactory.instance)
+    if (version >= 3) {
+      node.set("replicaId", new IntNode(request.replicaId))
+    }
+    val topicsToPartitionEpochs = CollectionUtils.groupPartitionDataByTopic(request.epochsByTopicPartition)
+    val topicsArray = new ArrayNode(JsonNodeFactory.instance)
+    for (topicToEpochs <- topicsToPartitionEpochs.entrySet.asScala) {
+      val topicsData = new ObjectNode(JsonNodeFactory.instance)
+      topicsData.set("name", new TextNode(topicToEpochs.getKey))
+      val partitionsArray = new ArrayNode(JsonNodeFactory.instance)
+      for (partitionEpoch <- topicToEpochs.getValue.entrySet.asScala) {
+        val partitionData = partitionEpoch.getValue
+        val partitionNode = new ObjectNode(JsonNodeFactory.instance)
+        partitionNode.set("partitionIndex", new IntNode(partitionEpoch.getKey))
+        partitionNode.set("leaderEpoch", new IntNode(partitionData.leaderEpoch))
+        if (version >= 2) {
+          val leaderEpoch = partitionData.currentLeaderEpoch
+          partitionNode.set("currentLeaderEpoch", new IntNode(leaderEpoch.orElse(RecordBatch.NO_PARTITION_LEADER_EPOCH)))
+        }
+        partitionsArray.add(partitionNode)
+      }
+      topicsData.set("partitions", partitionsArray)
+      topicsArray.add(topicsData)
+    }
+    node.set("topics", topicsArray)
+    node
+  }
+
+  /**
+   * Temporary until switch to use the generated schemas.
```

Comment:
I think that is fair, it would be good to align now. With the FetchResponse I would like to circle back on it and provide more detail. If not the recordSizeInBytes in the FetchResponse I think we need to include a top level responseSizeBytes in the trace.

Reply:
I've change it back to reflect what the auto-generated schema produces. The reason we didn't have it match the auto-generated schema was because it copies every byte of the record into the `records` field which could be expensive. If this isn't really a problem, I'll leave it as this.

Reply:
Actually, it is probably better to not serialize the records in our case. Records could be large and bytes are not that useful. I suppose that having the number of bytes as a field like we had before is likely way more valuable. Should we bring back the amount of bytes per partition for both the Fetch response and the Produce request? @lbradstreet What do you think?

Reply:
I agree that we shouldn't serialize the records. If it's easy to change the recordSet autogenerated JSON output I think it would be great to retain the sizes. This is something that I wanted to follow up with in a later PR but if we can do it now I think it would be very helpful.

Reply:
I changed the `recordSet` autogenerated JSON output to be the number of bytes, instead of an empty byte array.